### PR TITLE
SpriteFrameCache support async load

### PR DESCRIPTION
**Pull request by perminovVS**
Async based on Sprite3D and TextureCache.
Load texture and plist file in AsyncTaskPool TASK_IO thread.
Add TextureCache image and generate Sprite Frames in main thread.

### CODE REVIEW

File: cocos/2d/CCSpriteFrameCache.cpp [hunk 1]
```diff
@@ -435,4 +436,88 @@ SpriteFrame* SpriteFrameCache::getSpriteFrameByName(const std::string& name)
...
+		asyncLoadParam->image = nullptr;
+		asyncLoadParam->resultTexture = false;
+		asyncLoadParam->resultPList = false;
+
+
+		AsyncTaskPool::getInstance()->enqueue(AsyncTaskPool::TaskType::TASK_IO, CC_CALLBACK_1(SpriteFrameCache::afterAsyncLoad, this), (void*)(asyncLoadParam), [asyncLoadParam]()
+		{
+			// AsyncLoad call TASK_IO thread
+			std::string fullPathPList = FileUtils::getInstance()->fullPathForFilename(asyncLoadParam->plist);
+			std::string fullpathTexture = FileUtils::getInstance()->fullPathForFilename(asyncLoadParam->textureFileName);
```
Comment (minggo) [inline cocos/2d/CCSpriteFrameCache.cpp]: These two functions are not thread-safe.

**Comment by perminovVS** [comment]
fullPathForFilename now corrected

**Comment by minggo** [comment]
[ci rebuild]

**Comment by perminovVS** [comment]
@minggo Build successfully

**Comment by perminovVS** [comment]
@minggo merge please

**Comment by zilongshanren** [comment]
@perminovVS  
It seems this PR conflicts with the latest upstream, could you solve the merge conflicts?
Thanks.

**Comment by zilongshanren** [comment]
@perminovVS  Any updates?

**Comment by perminovVS** [comment]
@andyque Sorry, not enough time now to correct these days.

**Comment by perminovVS** [comment]
recreate in #12367

**Pull request closed without merging**